File: ecommerce/src/app/models/cartItem.ts
import { IProduct } from './product';
export class ICartItem{
    id: number;
    name: string;
    price: number;
    quantity: number;
    productId: number;
    shoppingCartId: number;
    product: IProduct;
    get totalPrice(){
        return this.quantity*this.product.price;
    }
}

File: Ecommerce.Model/EntityModels/CartItem.cs
using System;
using System.Collections.Generic;
using System.Reflection.Metadata;
using System.Text;
using Ecommerce.Model.EntityModels;

namespace Ecommerce.Model.HelpingModels
{
    public class CartItem
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public Product Product { get; set; }
        public int Quantity { get; set; }
        public int ShoppingCartId { get; set; }
        public ShoppingCart ShoppingCart { get; set; }
    }
}
